src/_guides/language/concurrency/index.md

```diff
@@ -0,0 +1,456 @@
+
+* ***[PENDING: link to example 4 in Isolate snippets][]***,
+  which shows how to send a message from
+  the main isolate to the spawned isolate.
+  It’s otherwise similar to the preceding example.
+* ***[PENDING: link to example 5 in Isolate snippets][]***,
+  which shows how to spawn a long-running isolate that
+  receives and sends multiple times.
+
+{{site.alert.tip}}
+  The `send()` method can be slow.
+  When possible,
+  avoid using `send()` to send large amounts of data.
+  Instead, send the data when the worker isolate exits,
+  as described in the previous section.
+{{site.alert.end}}
+
+## Performance and isolate groups
+
+When an isolate calls [`Isolate.spawn()`][],
+the two isolates have the same executable code
+and are in the same _isolate group_.
+Isolate groups enable performance optimizations such as sharing code;
+a new isolate immediately runs the code owned by the isolate group.
+Also, `Isolate.exit()` works only when the isolates
+are in the same isolate group.
+
+In some special cases —
+for example, when implementing tools like `dart pub run` —
+you might need to use [`Isolate.spawnUri()`][].
+That method sets up the new isolate with a copy of the code
+that's at the specified URI.
+However, `spawnUri()` is much slower than `spawn()`,
+and the new isolate isn't in its spawner's isolate group.
+
+[`Isolate.spawnUri()`]: https://api.dart.dev/stable/dart-isolate/Isolate/spawnUri.html
+
+
+
+-------------------
+
+## NOTES/QUESTIONS/TODOs (TO BE DELETED)
+
+TODO:
+* Before publishing for the first time:
+  * When examples 4 & 5 are published in dart-lang/samples
+    ([PR #128](https://github.com/dart-lang/samples/pull/128)),
+    link to them.
+    (If they aren't up by the time this page is published,
+    either remove the **Other examples** section or link to the PR.)
+  * Search for and resolve all PENDING items.
+  * Delete this section.
+* Before or soon after publishing:
+  * Decide whether to keep figure captions or delete them. Whatever we decide, format & use them consistently.
+  * Use macros in links to API & Flutter docs.
+  * Add alt text to figures.
+  * Use code excerpts.
+  * In text before figures, describe the figure content well enough to help people who translate the page or have accessibility issues.
+  * Add this page to the sidenav. Link to it appropriately.
+  * Figure out how to save an editable version of the source that has the right fonts. (The SVG files don't like the custom fonts; otherwise, I would've used SVGs instead of PNGs.)
+* Maybe:
+  * Add the following text somewhere in this page (or in the FAQ):
+    * Sometimes Dart is called a _single-threaded language._
+    * Dart code executes in a predictable sequence that can’t be interrupted by other Dart code.
+  * Add a figure up high in the doc? (if so, what?)
+
+Qs:
+* The figures are by the index file, under images. Should we instead use the asset system? E.g., see the source for the following figure:
```

Comment:
I think this is fine for now. I believe the rearch integration and related updates will remove support for the `{% asset` system.

Reply:
Good to know, thanks!